src/main/java/org/folio/search/controller/ErrorHandler.java

```diff
@@ -0,0 +1,23 @@
+package org.folio.search.controller;
+
+import static org.springframework.http.ResponseEntity.status;
+
+import org.folio.search.domain.dto.Error;
+import org.folio.search.domain.dto.ValidationErrors;
+import org.folio.search.exception.ValidationException;
+import org.folio.tenant.domain.dto.Parameter;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@RestControllerAdvice
+public class ErrorHandler {
+  @ExceptionHandler(ValidationException.class)
+  public ResponseEntity<ValidationErrors> handleValidationException(ValidationException ex) {
```

Comment:
this part of PR related to msearch-39

Reply:
Sure, but I have some tests that verifies that validation error is returned, they will fail if I remove this code.

Reply:
ok, Also I have a question, why do we need custom ValidationException, when Spring and Java provide other options? like IllegalArgumentException or ConstraintViolationException

Reply:
I'm fine using existing exceptions. What exception are you suggesting for validation? `ConstraintViolationException` looks quite tricky to construct in order to be able to return the FOLIO standard validation error response, i.e.:
```
{
  "message": "..."
  "parameters": [{
     "key": "the caused property name",
     "value": "the value caused the issue"
  }]
}
```